src/credentials/GroupDataProvider.h

```diff
@@ -138,6 +147,31 @@ class GroupDataProvider
             VerifyOrReturnError(this->policy == other.policy && this->num_keys_used == other.num_keys_used, false);
             return !memcmp(this->epoch_keys, other.epoch_keys, this->num_keys_used * sizeof(EpochKey));
         }
+
+        ByteSpan GetCurrentKey()
+        {
+            auto now                                = System::SystemClock().GetMonotonicTimestamp();
```

Comment:
This is problematic since the monotonic timestamp is not real/wall time, but epochs are based on wall time.

There is mention in the spec of what rules to employ when time is not synchronized. I suggest we use those rules *by default* now (assume time is synchronized on nobody) and add a TODO about handling time synchronized versus not, since this is already not very well supported in the SDK, especially since the time sync cluster is not implemented fully and not mandatory.

Reply:
Using EpochKey0 now, which is the active `Current` key.